py/desiutil/install.py

```diff
@@ -847,7 +854,11 @@ def install(self):
                         os.chdir(self.install_dir)
                     else:
                         os.chdir(self.working_dir)
-                    proc = Popen(command, universal_newlines=True,
+                    if 'cmake' in self.build_type:                        
+                        proc = Popen(command, universal_newlines=True,
+                                 stdout=PIPE, stderr=PIPE, shell=True)
```

Comment:
This seems messy / fragile where "cmake" requires two-commands-in-one and thus a string command with `shell=True`, while "make" uses a single-command-passed-as-a-list with default `shell=False`.  The normal admonishment against `shell=True` [[link](https://docs.python.org/3/library/subprocess.html#security-considerations)] is due to security concerns about shell injection vulnerabilities from unsanitized user input strings, but in this case we have completely control over the commands created without any user-provided options, and thus I think both "make" and "cmake" could use string (not list) commands with `shell=True` to avoid this if/else.  Thoughts?

Reply:
To further @sbailey's comment, did you actually *test* the commands with `shell=False` and come to the conclusion that it was impossible to run with `shell=False`?

I don't think anyone would object if we ran two commands, both with `shell=False`, if the one and only reason for `shell=True` is a `;`.

Reply:
Did a test and found it would require two running two commands and the `shell=True` was chosen. Will try the `setup.py` solution @tskisner mentioned below and if so these changes will be unnecessary.

Reply:
That sounds promising.  I was about to point out that you have `if 'cmake' in self.build_type:`, but you have not modified the code in any way to *detect* `'cmake'` as the `build_type`.